chapters/intro-computer-architecture/from-code-to-executable/reading/README.md

```diff
@@ -0,0 +1,142 @@
+# From Code to Executable
+
+The steps through which a `C` source code goes, from the moment of writing the code up until running as a process, are the following:
+
+- preprocesing
+- compiling
+- assembly
+- linking
+
+The following diagram illustrates the steps:
+
+TODO
+
+## Preprocessing
+
+During this step, the following actions take place, starting from the source code:
+* comment removal
+* expansion of preprocessor directives (the ones that begin with `#`), like `#define`, `#include`, `#pragma`.
+
+With `gcc`, this step can be run using the `-E` flag.
+The output file will have the `.i` extension.
+```bash
+$ gcc -E main.c -o main.i
```

Comment:
```suggestion
student@hsi:~$ gcc -E main.c -o main.i
```
Add generic prompts whenever there's a command that's not supposed to be run from a specific folder (such as a task or a guide).